src/main/java/it/geosolutions/httpproxy/HTTPProxy.java

```diff
@@ -786,15 +787,13 @@ else if (header.getName().equalsIgnoreCase(Utils.HTTP_HEADER_TRANSFER_ENCODING))
             if(inputStreamServerResponse != null){
                 byte[] b = new byte[proxyConfig.getDefaultStreamByteSize()];
                 
-                baos = new ByteArrayOutputStream(b.length);
-                
                 int read = 0;
+                ServletOutputStream out = httpServletResponse.getOutputStream();
     		    while((read = inputStreamServerResponse.read(b)) > 0){ 
-    		      	baos.write(b, 0, read);
-    		        baos.flush();
+    		      	out.write(b, 0, read);
+    		      	out.flush();
```

Comment:
Not sure this is correct, it is up to the Servlet Container to decide when doing this and anyway, I would not flush every 1KB.

At the very least, we might want to also increase the default size of the b array. Right now is 1KB I would go at least to 8KB if not 64KB.

Notice that the underlying streams are usually buffered.